src/Storage/Services/RegisterService.cs

```diff
@@ -103,6 +105,14 @@ public async Task<int> PartyLookup(string person, string orgNo)
                 Party party = await response.Content.ReadFromJsonAsync<Party>(_serializerOptions);
                 return party.PartyId;
             }
+            else if (response.StatusCode == HttpStatusCode.NotFound)
+            {
+                string reason = await response.Content.ReadAsStringAsync();
+                _logger.LogError("// RegisterService // PartyLookup // Party not found. Response status code is {StatusCode}. \n Reason {Reason}.", response.StatusCode, reason);
+
+                // Return a custom exception indicating that the party was not found
+                throw await PartyNotFoundException.CreateAsync(response);
```

Comment:
It's tempting to simplify this down to `return null;`

This is not an error or "exceptional". It's going to happen somewhat regularly.

There are two possible reasons for logging this:
1. We need statistics. How often are we given a person or organization that doesn't exist.
2. We need to analyze a support case. Someone is trying to download instances on an organization that is unknown to Altinn.